### ISSUE TITLE
Bad request exception on commit before persisting anything

### ISSUE THREAD

**Issue by imvaid** [ongr-io/ElasticsearchBundle#356]
If we try to commit before persisting anything (as we have situation when we do not always know that we persisted anything), we get bad request exception:
 ```
ElasticsearchParseException[Failed to derive xcontent from org.elasticsearch.common.bytes.BytesArray@1]
```

Error occurs when the following line returns empty result and we try to call bulk query with empty body:
https://github.com/ongr-io/ElasticsearchBundle/blob/9b18076e1af8f47fa61cc5e659effc6b3630ced4/Client/Connection.php#L156

Consider adding result check before calling bulk query or maybe throw custom exception?

**Comment by tautrimas** [comment]
:+1: This is an issue in many bundles where nasty workarounds are being deployed.

**Comment by saimaz** [comment]
Its already implemented in #366

**Comment by tautrimas** [comment]
Wouldn't it be better to add argument, so that it silently ignores the error altogether? It's tedious to keep track whether there was a single document persisted or not. We just need to ensure everything is `commit`ed and forget.

**Comment by tautelis** [comment]
This issue must be pretty common. Any updates on it?

**Comment by murnieza** [comment]
Still no progress on this one?

**Comment by saimaz** [comment]
Not yet, it should be fixed latest September 1st, might be even earlier.

**Comment by saimaz** [comment]
Solved in #405
